src/OpenLoco/src/Localisation/FormatArguments.hpp

```diff
@@ -61,26 +59,27 @@ namespace OpenLoco
 
         FormatArguments()
         {
-            loco_global<std::byte[20], 0x0112C826> _commonFormatArgs;
-
-            _bufferStart = _buffer = &*_commonFormatArgs;
-            _capacity = std::size(_commonFormatArgs);
+            // TODO: refactor users to use non-static buffers
+            static std::byte defaultBuffer[20];
```

Comment:
i should probably know this but does this work? This is a header so every instance surely is a new static buffer.

Reply:
I was somewhat pleasantly surprised this works, too. It has no unexpected side effects for me this way, i.e. making the buffers static in the function bodies.

If I made the buffers static _outside_ the functions, as I had done initially, they _would_ be different for each unit, though (breaking, e.g., the map tooltips).

Reply:
I've been having a read online and that might be true for non templated functions but the map tooltip one is templated so there will be multiple globals for the various instances.

Reply:
The cheap way to do it would be to make a non templated function that sets the buffer (like how common buffer works). I still don't especially like the solution but it's the minimum work.

Reply:
I wonder if it wasn't an issue because the template was using variadic parameters? Anyway, I've gone with the solution you suggested.

Reply:
> Inline would only have one instance?? That's really counter-intuitive to me, given it's a header file and all that.
> 

That has been added in C++17, inline variables and yes its an odd choice, don't blame the messenger.

> Problem there is that `FormatArgumentsBuffer` is currently always 16 bytes. This is not enough to store the map tooltip arguments. It's currently 40 bytes, which might be excessively long, but I was getting buffer position exceptions when trying to make it work with just 16 bytes.

Nothing stops us from increasing it to 40 or even bigger if needed, I think majority that uses it is already interop free.